crates/shim/src/cgroup.rs

```diff
@@ -199,6 +189,37 @@ fn get_cgroup(pid: u32) -> Result<Cgroup> {
     Ok(cgroup)
 }
 
+/// Get the cgroups v2 path given a PID
+pub fn get_cgroups_v2_path_by_pid(pid: u32) -> Result<String> {
+    // todo: should upstream to cgroups-rs
+    let path = format!("/proc/{}/cgroup", pid);
+    let content = fs::read_to_string(path).map_err(io_error!(e, "read cgroup"))?;
+    let content = content.trim_end_matches('\n');
```

Comment:
**question:** What should this do in the case of multiple lines in the file? Not even sure that's possible though :-)
**suggestion:**
```rust
let content = content.lines().next().unwrap_or("");
```